engine/include/cubos/engine/input/bindings.hpp

```diff
@@ -0,0 +1,115 @@
+/// @file
+/// @brief
+
+#pragma once
+
+#include <optional>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <cubos/core/io/keyboard.hpp>
+
+namespace cubos::engine
+{
+    class KeyWithModifier final
+    {
+    public:
+        KeyWithModifier() = default;
```

Comment:
In this case it's not very clear what would the default value be - it is probably 0, which is `Key::A`, when it should be `Key::Invalid` - I would prefer you either define the constructor and the key to `Key::Invalid`, or add a field initializer down in the private section, like
```cpp
core::io::Key mKey = core::io::Key::Invalid;
```